docs/integration/ldap.md

```diff
@@ -0,0 +1,141 @@
+---
+tags:
+  - on-premise
+  - auth
+---
+
+# How to use LDAP for Authentication
+
+Galaxy can use `django_auth_ldap` package to enable LDAP integration, more info about the plugin can
+be found on the plugin docs at: https://django-auth-ldap.readthedocs.io/ 
+
+## Requirements
+
+A running and acessible `LDAP` or `AD` server.
+
+!!! tip
+    For testing purposes you can start an openldap testing server using
+    `https://github.com/rroemhild/docker-test-openldap` this server runs on `10389` port.
+
+
+Checking if your ldap server is up and running:
+
+> You need ldap-utils installed
+
+```bash
+# List all users
+ldapsearch -H ldap://localhost:10389 -x -b "ou=people,dc=planetexpress,dc=com" -D "cn=admin,dc=planetexpress,dc=com" -w GoodNewsEveryone "(objectClass=inetOrgPerson)"
+
+# List all groups
+ldapsearch -H ldap://localhost:10389 -x -b "ou=people,dc=planetexpress,dc=com" -D "cn=admin,dc=planetexpress,dc=com" -w GoodNewsEveryone "(objectClass=Group)"
+```
+
+## Enabling Galaxy LDAP integration
+
+The following settings can be added to either `/etc/pulp/settings.py` or exported as environment
+variables prefixed with `PULP_`.
+
+
+Example using environment variables:
+
+!!! tip
+    To set those variables using `/etc/pulp/settings.py` remove `PULP_` prefix and instead of toml formatting
+    declare as normal python objects such as bools, lists and dictionaries.
+
+
+Authentication class and deployment mode by default is already set tho the following values, 
+You don't need to change it, just confirm this is the setting you have in place.
+
+```bash
+PULP_GALAXY_AUTHENTICATION_CLASSES=['rest_framework.authentication.SessionAuthentication','rest_framework.authentication.TokenAuthentication','rest_framework.authentication.BasicAuthentication']
+PULP_GALAXY_DEPLOYMENT_MODE=standalone
+```
+
+Pulp container requires this to be set in order to provide docker registry compatible token authentication.
+https://docs.pulpproject.org/pulp_container/authentication.html
+
+```bash
+PULP_TOKEN_AUTH_DISABLED=true
+```
+
+`django_auth_ldap` must be included as the first authentication backend.
+
+```bash
+PULP_AUTHENTICATION_BACKENDS=["django_auth_ldap.backend.LDAPBackend","django.contrib.auth.backends.ModelBackend","guardian.backends.ObjectPermissionBackend","pulpcore.backends.ObjectRolePermissionBackend"]
```

Comment:
I'm not sure it's a good idea to expose this configuration option directly to the user. If they mess up `pulpcore.backends.ObjectRolePermissionBackend` for example, the app will break spectacularly. Could we create a `GALAXY_AUTHENTICATION_PRESET` setting (or something like that), that accepts `local`, `ldap` and maybe `keycloak` and configures the django settings under the hood?